optaplanner-wb-screens/optaplanner-wb-guided-rule-editor/optaplanner-wb-guided-rule-editor-backend/src/main/java/org/optaplanner/workbench/screens/guidedrule/backend/server/plugin/PersistenceExtensionUtils.java

```diff
@@ -18,17 +18,29 @@
 
 public final class PersistenceExtensionUtils {
 
+    public static final String EXCEPTION_MESSAGE_BASE = "Could not unmarshal action string ";
+
     private PersistenceExtensionUtils() {
     }
 
-    public static String unwrapParenthesis(final String s) {
-        int start = s.indexOf('(');
-        int end = s.lastIndexOf(')');
+    private static String unwrap(final String stringToUnwrap,
+                                 final Character wrappingCharacterStart,
+                                 final Character wrappingCharacterEnd) {
+        int start = stringToUnwrap.indexOf(wrappingCharacterStart);
+        int end = stringToUnwrap.lastIndexOf(wrappingCharacterEnd);
         if (start < 0 || end < 0) {
-            return s;
+            return stringToUnwrap;
         }
-        return s.substring(start + 1,
-                           end).trim();
+        return stringToUnwrap.substring(start + 1,
+                                        end).trim();
```

Comment:
I'd prefer to add a parameter validation check that would make sure that `start < end` that would give an illegal argument exception with a meaningful message instead of StringIndexOutOfBoundsException.

Another validation check suggestion: start char should be different from end char.

Reply:
Definitely I can apply.